js/_api.js

```diff
@@ -10,31 +11,74 @@ export function APIError(message, json) {
 APIError.prototype = Object.create(Error.prototype);
 APIError.prototype.constructor = APIError;
 
-function sleep(time) {
+// SETUP_API is a cross dispatcher action that stores information recieved
+// from the backend dispatcher so that the api class can dip into the store and
+// find the correct endpoints and csrf tokens depending on the namespace
+export const SET_API = 'SET_API';
+export function setAPI(namespace, api_endpoint, api_auth) {
+	return {
+		type: SET_API,
+		namespace: namespace,
+		endpoint: api_endpoint,
+		auth: api_auth
+	};
+}
+
+export function sleep(time) {
 	return new Promise((resolve) => setTimeout(resolve, time));
 }
 
-export function call(uri, method, payload) {
+var namespace = '';
+export function setNamespace(ns) {
+	namespace = ns;
+}
+
+export function call(dispatch, getState, uri, method, payload) {
```

Comment:
Since the CSRF token, api endpoint and other dispatcher specific information is in the store, we need to dispatch new information when it's recieved and also being able to get the state so we can include the CSRF in the request.